lib/TH/vector/AVX.c

```diff
@@ -123,6 +136,19 @@ static void THDoubleVector_add_AVX(double *y, const double *x, const double c, c
   }
 }
 
+static void THFloatVector_copy_AVX(float *y, const float *x, const ptrdiff_t n) {
+  ptrdiff_t i;
+  ptrdiff_t off;
+  for (i=0; i<=((n)-16); i+=16) {
+    _mm256_storeu_ps(y+i, _mm256_loadu_ps(x+i));
+    _mm256_storeu_ps(y+i+8, _mm256_loadu_ps(x+i+8));
```

Comment:
What do you mean by "stream instructions"?
If you want to do aligned instructions, can't you just check if the pointers are aligned?

```
if (!(x & 0x10) && !(y & 0x10)) {
  // do the aligned thing
} else {
  // something else
}
```